app/src/main/addressbook.js

```diff
@@ -139,7 +144,11 @@ module.exports = class Addressbook {
     let subPeers = (await axios.get(
       `http://${peerIP}:${this.config.default_tendermint_port}/net_info`
     )).data.result.peers
-    let subPeersHostnames = subPeers.map(peer => peer.node_info.listen_addr)
+
+    let subPeersHostnames = subPeers
```

Comment:
this is only excluding non indexing nodes in the discovery phase. the flag could skip in the meantime until we connect to that node. I suggest we check the flag when we are trying to connect to the node.

Reply:
you mean when adding the peer ?

Reply:
No, when connecting to a peer. Probably here https://github.com/cosmos/voyager/blob/35fa2baa86e9943097d43b5507616273bffeb2a1/app/src/main/addressbook.js#L98

Reply:
`this.peers` is just the `hostname` and a `state`, it doesn't show the full peer `node_info`. Maybe we should consider adding a new `boolean` attribute called `indexing` to each `peer` Object and pass it to the `addPeer` function

Reply:
So this are the options:
1) leave the `isIndexing` filter on the `discoverPeers` function
2) change the whole Addressbook logic to be able to filter peers on the `pickNode` function

Reply:
As I wrote before: the indexing behavior could change in the time between adding the peer and connecting to the peer.

1) therefor is not an option

would you disagree?